### PR TITLE
Add ReseNetE18 and BinaryDenseNet{28,37,45}

### PR DESCRIPTION
**Pull request by lgeiger**
This adds the DenseNet28 and DenseNet45 models from https://arxiv.org/abs/1906.08637

~~Note: This PR is build on top of #59 so only the last two commits are relevant for review here.~~

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -131,6 +131,50 @@ A Keras model instance.
 
 **References**
 
+- [Back to Simplicity:
+  How to Train Accurate BNNs from Scratch?](https://arxiv.org/abs/1906.08637)
+
+## DenseNet28<a class="headerlink code-link" style="float:right;" href="https://github.com/larq/zoo/blob/master/larq_zoo/densenet.py#L119" title="Source code"></a>
+
+```python
+DenseNet28(include_top=True,
```
Comment (timdebruin) [inline docs/api.md]: To avoid ambiguity when comparing full precision architectures should we call this BinaryDensenet? Of course we could `import Densenet as BinaryDensenet` but being explicit (and calling it exactly as in the paper) might prevent some confusion.

File: larq_zoo/densenet.py [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        use_bias=False,
+    )(y)
+    return keras.layers.concatenate([x, y])
+
+
+@registry.register_model
+def densenet(args, dataset, input_tensor=None, include_top=True):
+    input = utils.get_input_layer(dataset.input_shape, input_tensor)
+
+    if dataset.input_shape[0] and dataset.input_shape[0] < 50:
```
Comment (timdebruin) [inline larq_zoo/densenet.py]: Same as above

File: larq_zoo/densenet.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        input_quantizer=args.quantizer,
+        kernel_quantizer=args.quantizer,
+        kernel_initializer="glorot_normal",
+        kernel_constraint=args.constraint,
+        padding="same",
+        use_bias=False,
+    )(y)
+    return keras.layers.concatenate([x, y])
+
+
```
Comment (MariaHeuss) [inline larq_zoo/densenet.py]: Maybe it would be good to have a comment somewhere on how the imageshape of the dataset should look like for all the models we have. Especially if there are different possible image sizes as here.

File: larq_zoo/densenet.py (hunk 2 above)
Comment (MariaHeuss) [inline larq_zoo/densenet.py]: Maybe it would be good to have a comment somewhere on how the imageshape of the dataset should look like for all the models we have. Especially if there are different possible image sizes as here.

File: larq_zoo/densenet.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+@registry.register_hparams(densenet)
+class densenet37(densenet28):
+    batch_size = 192
+    learning_steps = [80, 100]
+    reduction = [3.3, 3.3, 4]
+    layers = [6, 8, 12, 6]
+
+
+@registry.register_hparams(densenet)
+class densenet45(densenet37):
```
Comment (MariaHeuss) [inline larq_zoo/densenet.py]: would not make more sense to inherit from densenet28 again?

File: larq_zoo/densenet.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        kernel_initializer="glorot_normal",
+        kernel_constraint=args.constraint,
+        padding="same",
+        use_bias=False,
+    )(y)
+    return keras.layers.concatenate([x, y])
+
+
+@registry.register_model
+def densenet(args, dataset, input_tensor=None, include_top=True):
```
[the remark this answers is not in the document]
Comment (lgeiger) [reply larq_zoo/densenet.py]: Rebased it 👍

File: larq_zoo/densenet.py (hunk 2 above)
Comment (lgeiger) [reply larq_zoo/densenet.py]: It is mentioned in the docstrings. This line is mostly in order to allow training with CIFAR10

File: docs/api.md (hunk 1 above)
Comment (lgeiger) [reply docs/api.md]: I agree. That seams like a good idea.
